Source/Core/Core/HW/DSPHLE/UCodes/UCodes.cpp

```diff
@@ -282,6 +283,13 @@ std::unique_ptr<UCodeInterface> UCodeFactory(u32 crc, DSPHLE* dsphle, bool wii)
     INFO_LOG_FMT(DSPHLE, "CRC {:08x}: Wii - AXWii chosen", crc);
     return std::make_unique<AXWiiUCode>(dsphle, crc);
 
+  case 0x8d69a19b:
+  case 0xc2fd441c:
```

Comment:
I couldn't get the old one to boot even with messing around with CPU Clock Override, but building your version with the newest libraries worked fine. The way the old version fails to boot seems to be the same as Rick Dangerous Wii (see [issue 12799](https://bugs.dolphin-emu.org/issues/12799)); either of the patches I made for that also fixes this, although patch B seems more crackly.

The CRC from the original (0xcc2fd441) is in my big list of possible CRCs in https://github.com/dolphin-emu/dolphin/pull/10763#issuecomment-1159613562 as `3/dsp_mixer_padded.bin` (while the implemented one (0xc2fd441c), which looks similar because the hash being used here is kinda terrible, is `3/dsp_mixer.bin`). Perhaps there's other padding that happens in a place other than just the header file, in which case the unpadded version wouldn't be used at all.

I also did get a bit of a loud static sound with the recompiled version for the sound of something coming out of a ? block (with or without the patches from issue 12799). I hear something funky in the original version, so perhaps the issue existed in the original version but things were laid out differently in memory resulting in no loud static?

Reply:
OK, I've determined that it should be padded - [the array in the header](https://github.com/devkitPro/libogc/blob/3b1f018dbe372859a43bff8560e2525f6efa4433/libasnd/dsp_mixer.h) has 726 (0x2d6) entries (i.e. 1452 (0x5ac) bytes), but `size_dsp_mixer` is 1472 (0x5c0), so the number of elements in `dsp_mixer`, which is `size_dsp_mixer/2`, is 736 (0x2e0) and thus the remaining entries are zero-initialized (I think). Thus the array _is_ padded in this case, even though it wasn't padded in the header itself. So the CRC of `0xcc2fd441` is the correct one, and `0xc2fd441c` shouldn't actually be used by anything.

The most recent version, which doesn't use a header but instead uses `bin2o`, is not padded (and has a hash of `0xdbbeeb61`), as I determined before.

Thanks for pointing this out.